-diff --git a/arch/arm/boot/dts/qcom/kenzo/msm8956-kenzo.dtsi b/arch/arm/boot/dts/qcom/kenzo/msm8956-kenzo.dtsi
-index 5677a0b..c7cb705 100644
---- a/arch/arm/boot/dts/qcom/kenzo/msm8956-kenzo.dtsi
-+++ b/arch/arm/boot/dts/qcom/kenzo/msm8956-kenzo.dtsi
-@@ -231,12 +231,12 @@
- 			< 133333333   1 >,  /* Min SVS   */
- 			< 200000000   2 >,  /* Low SVS   */
- 			< 266666667   3 >,  /* SVS Minus */
--			< 300000000   4 >,  /* SVS       */
--			< 366670000   5 >,  /* SVS Plus  */
--			< 432000000   6 >,  /* NOM       */
--			< 480000000   7 >,  /* Turbo     */
--			< 550000000   8 >,  /* Super Turbo */
--			< 600000000   9 >;  /* Super Turbo */
-+			< 366670000   4 >,  /* SVS       */
-+			< 432000000   5 >,  /* SVS Plus  */
-+			< 480000000   6 >,  /* NOM       */
-+			< 550000000   7 >,  /* Turbo     */
-+			< 652800000   8 >,  /* Super Turbo */
-+			< 883200000   9 >;  /* Super Turbo */
- 		#clock-cells = <1>;
- 	};
- 
-@@ -316,8 +316,8 @@
- 			< 1190400000 5>,
- 			< 1305600000 6>,
- 			< 1382400000 7>,
--			< 1401600000 8>,
--			< 1478400000 9>;
-+			< 1478400000 8>,
-+			< 1708800000 9>;
- 		qcom,speed0-bin-v0-c1 =
- 			<          0 0>,
- 			<  400000000 1>,
-@@ -325,8 +325,8 @@
- 			< 1190400000 3>,
- 			< 1382400000 4>,
- 			< 1612800000 5>,
--			< 1747200000 6>,
--			< 1843200000 7>;
-+			< 1843200000 6>,
-+			< 2380800000 7>;
- 		qcom,speed0-bin-v0-cci =
- 			<          0 0>,
- 			<  307200000 1>,
-@@ -343,8 +343,8 @@
- 			< 1190400000 5>,
- 			< 1305600000 6>,
- 			< 1382400000 7>,
--			< 1401600000 8>,
--			< 1478400000 9>;
-+			< 1478400000 8>,
-+			< 1708800000 9>;
- 		qcom,speed1-bin-v0-c1 =
- 			<          0 0>,
- 			<  400000000 1>,
-@@ -352,8 +352,8 @@
- 			< 1190400000 3>,
- 			< 1382400000 4>,
- 			< 1612800000 5>,
--			< 1747200000 6>,
--			< 1843200000 7>;
-+			< 1843200000 6>,
-+			< 2380800000 7>;
- 		qcom,speed1-bin-v0-cci =
- 			<          0 0>,
- 			<  307200000 1>,
-@@ -439,13 +439,13 @@
- 				<  806400  4101 >,	/* SVS+  */
- 				< 1017600  5712 >,	/* NOM   */
- 				< 1190400  6152 >,	/* TURBO */
--				< 1478400  7104 >;
-+				< 1708800  7104 >;
- 			cpu-to-dev-map-4 =
- 				<  883200   805 >,	/* SVS   */
- 				< 1190400  4101 >,	/* SVS+  */
- 				< 1382400  5712 >,	/* NOM   */
- 				< 1612800  6152 >,	/* TURBO */
--				< 1843200  7104 >;
-+				< 2380800  7104 >;
- 		};
- 
- 		mincpubw-cpufreq {
-@@ -467,13 +467,13 @@
- 				<  806400  403200 >,
- 				< 1017600  441600 >,
- 				< 1190400  556800 >,
--				< 1478400  614400 >;
-+				< 1708800  614400 >;
- 			cpu-to-dev-map-4 =
- 				<  883200 307200 >,
- 				< 1190400 403200 >,
- 				< 1382400 441600 >,
- 				< 1612800 556800 >,
--				< 1843200 614400 >;
-+				< 2380800 614400 >;
- 		};
- 	};
- 
-@@ -498,7 +498,8 @@
- 			< 1305600 >,
- 			< 1382400 >,
- 			< 1401600 >,
--			< 1478400 >;
-+			< 1478400 >,
-+			< 1708800 >;
- 		qcom,cpufreq-table-4 =
- 			<  400000 >,
- 			<  883200 >,
-@@ -512,7 +513,8 @@
- 			< 1382400 >,
- 			< 1612800 >,
- 			< 1747200 >,
--			< 1843200 >;
-+			< 1843200 >,
-+			< 2380800 >;
- 	};
- 
- 	timer {
-diff --git a/arch/arm/boot/dts/qcom/msm8956.dtsi b/arch/arm/boot/dts/qcom/msm8956.dtsi
-index 36f9d774..d4f2295 100644
---- a/arch/arm/boot/dts/qcom/msm8956.dtsi
-+++ b/arch/arm/boot/dts/qcom/msm8956.dtsi
-@@ -225,12 +225,12 @@
- 			< 133333333   1 >,  /* Min SVS   */
- 			< 200000000   2 >,  /* Low SVS   */
- 			< 266666667   3 >,  /* SVS Minus */
--			< 300000000   4 >,  /* SVS       */
--			< 366670000   5 >,  /* SVS Plus  */
--			< 432000000   6 >,  /* NOM       */
--			< 480000000   7 >,  /* Turbo     */
--			< 550000000   8 >,  /* Super Turbo */
--			< 600000000   9 >;  /* Super Turbo */
-+			< 366670000   4 >,  /* SVS       */
-+			< 432000000   5 >,  /* SVS Plus  */
-+			< 480000000   6 >,  /* NOM       */
-+			< 550000000   7 >,  /* Turbo     */
-+			< 652800000   8 >,  /* Super Turbo */
-+			< 883200000   9 >;  /* Super Turbo */
- 		#clock-cells = <1>;
- 	};
- 
-@@ -310,8 +310,8 @@
- 			< 1190400000 5>,
- 			< 1305600000 6>,
- 			< 1382400000 7>,
--			< 1401600000 8>,
--			< 1478400000 9>;
-+			< 1478400000 8>,
-+			< 1708800000 9>;
- 		qcom,speed0-bin-v0-c1 =
- 			<          0 0>,
- 			<  400000000 1>,
-@@ -319,8 +319,8 @@
- 			< 1190400000 3>,
- 			< 1382400000 4>,
- 			< 1612800000 5>,
--			< 1747200000 6>,
--			< 1843200000 7>;
-+			< 1843200000 6>,
-+			< 2380800000 7>;
- 		qcom,speed0-bin-v0-cci =
- 			<          0 0>,
- 			<  307200000 1>,
-@@ -337,8 +337,8 @@
- 			< 1190400000 5>,
- 			< 1305600000 6>,
- 			< 1382400000 7>,
--			< 1401600000 8>,
--			< 1478400000 9>;
-+			< 1478400000 8>,
-+			< 1708800000 9>;
- 		qcom,speed1-bin-v0-c1 =
- 			<          0 0>,
- 			<  400000000 1>,
-@@ -346,8 +346,8 @@
- 			< 1190400000 3>,
- 			< 1382400000 4>,
- 			< 1612800000 5>,
--			< 1747200000 6>,
--			< 1843200000 7>;
-+			< 1843200000 6>,
-+			< 2380800000 7>;
- 		qcom,speed1-bin-v0-cci =
- 			<          0 0>,
- 			<  307200000 1>,
-@@ -431,13 +431,13 @@
- 				<  806400  4101 >,	/* SVS+  */
- 				< 1017600  5712 >,	/* NOM   */
- 				< 1190400  6152 >,	/* TURBO */
--				< 1478400  7104 >;
-+				< 1708800  7104 >;
- 			cpu-to-dev-map-4 =
- 				<  883200   805 >,	/* SVS   */
- 				< 1190400  4101 >,	/* SVS+  */
- 				< 1382400  5712 >,	/* NOM   */
- 				< 1612800  6152 >,	/* TURBO */
--				< 1843200  7104 >;
-+				< 2380800  7104 >;
- 		};
- 
- 		mincpubw-cpufreq {
-@@ -459,13 +459,13 @@
- 				<  806400  403200 >,
- 				< 1017600  441600 >,
- 				< 1190400  556800 >,
--				< 1478400  614400 >;
-+				< 1708800  614400 >;
- 			cpu-to-dev-map-4 =
- 				<  883200 307200 >,
- 				< 1190400 403200 >,
- 				< 1382400 441600 >,
- 				< 1612800 556800 >,
--				< 1843200 614400 >;
-+				< 2380800 614400 >;
- 		};
- 	};
- 
-@@ -490,7 +490,8 @@
- 			< 1305600 >,
- 			< 1382400 >,
- 			< 1401600 >,
--			< 1478400 >;
-+			< 1478400 >,
-+			< 1708800 >;
- 		qcom,cpufreq-table-4 =
- 			<  400000 >,
- 			<  883200 >,
-@@ -504,7 +505,8 @@
- 			< 1382400 >,
- 			< 1612800 >,
- 			< 1747200 >,
--			< 1843200 >;
-+			< 1843200 >,
-+			< 2380800 >;
- 	};
- 
- 	timer {
-diff --git a/arch/arm/boot/dts/qcom/msm8976-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8976-gpu.dtsi
-index 31429fa..e9aa9cd 100644
---- a/arch/arm/boot/dts/qcom/msm8976-gpu.dtsi
-+++ b/arch/arm/boot/dts/qcom/msm8976-gpu.dtsi
-@@ -132,7 +132,7 @@
- 			/* TURBO */
- 			qcom,gpu-pwrlevel@0 {
- 				reg = <0>;
--				qcom,gpu-freq = <600000000>;
-+				qcom,gpu-freq = <883200000>;
- 				qcom,bus-freq = <10>;
- 				qcom,bus-min = <9>;
- 				qcom,bus-max = <10>;
-@@ -141,7 +141,7 @@
- 			/* TURBO */
- 			qcom,gpu-pwrlevel@1 {
- 				reg = <1>;
--				qcom,gpu-freq = <550000000>;
-+				qcom,gpu-freq = <652800000>;
- 				qcom,bus-freq = <9>;
- 				qcom,bus-min = <9>;
- 				qcom,bus-max = <10>;
-@@ -150,7 +150,7 @@
- 			/* NOM+ */
- 			qcom,gpu-pwrlevel@2 {
- 				reg = <2>;
--				qcom,gpu-freq = <480000000>;
-+				qcom,gpu-freq = <550000000>;
- 				qcom,bus-freq = <8>;
- 				qcom,bus-min = <7>;
- 				qcom,bus-max = <9>;
-@@ -159,7 +159,7 @@
- 			/* NOM */
- 			qcom,gpu-pwrlevel@3 {
- 				reg = <3>;
--				qcom,gpu-freq = <432000000>;
-+				qcom,gpu-freq = <480000000>;
- 				qcom,bus-freq = <7>;
- 				qcom,bus-min = <6>;
- 				qcom,bus-max = <8>;
-@@ -168,7 +168,7 @@
- 			/* SVS */
- 			qcom,gpu-pwrlevel@4 {
- 				reg = <4>;
--				qcom,gpu-freq = <300000000>;
-+				qcom,gpu-freq = <366670000>;
- 				qcom,bus-freq = <6>;
- 				qcom,bus-min = <5>;
- 				qcom,bus-max = <7>;
-diff --git a/arch/arm/boot/dts/qcom/msm8976-regulator.dtsi b/arch/arm/boot/dts/qcom/msm8976-regulator.dtsi
-index 96c9950..cb076cb 100644
---- a/arch/arm/boot/dts/qcom/msm8976-regulator.dtsi
-+++ b/arch/arm/boot/dts/qcom/msm8976-regulator.dtsi
-@@ -452,8 +452,8 @@
- 				<5  1190400000>,
- 				<6  1305600000>,
- 				<7  1382400000>,
--				<8  1401600000>,
--				<9  1478400000>;
-+				<8  1478400000>,
-+				<9  1708800000>;
- 		qcom,cpr-speed-bin-max-corners =
- 				<0 0 2 4 8>,
- 				<1 0 2 4 9>;
-@@ -470,7 +470,7 @@
- 			<0  0   0>;
- 		qcom,cpr-voltage-floor-override =
- 			<(-1) (-1) 765000 765000 805000 825000
--			915000 915000 960000 960000 970000>;
-+			915000 915000 960000 960000 980000>;
- 		qcom,cpr-enable;
- 	};
- 
-@@ -547,8 +547,8 @@
- 				<3  1190400000>,
- 				<4  1382400000>,
- 				<5  1612800000>,
--				<6  1747200000>,
--				<7  1843200000>;
-+				<6  1843200000>,
-+				<7  2380800000>;
- 		qcom,cpr-speed-bin-max-corners =
- 				<0 0 2 4 6>,
- 				<1 0 2 4 7>;
-@@ -565,7 +565,7 @@
- 			<0    0  0>;
- 		qcom,cpr-voltage-floor-override =
- 			<(-1) (-1) 760000 780000 790000 820000
--			910000 960000 970000>;
-+			910000 960000 980000>;
- 		qcom,cpr-enable;
- 	};
- 
-diff --git a/drivers/clk/qcom/clock-cpu-8976.c b/drivers/clk/qcom/clock-cpu-8976.c
-index 130314d..4690a2e 100644
---- a/drivers/clk/qcom/clock-cpu-8976.c
-+++ b/drivers/clk/qcom/clock-cpu-8976.c
-@@ -141,7 +141,7 @@ static struct pll_clk a72ss_hf_pll = {
- 		.vco_mode_masked = BVAL(21, 20, 1),
- 	},
- 	.base = &virt_bases[APCS_C1_PLL_BASE],
--	.max_rate = 1843200000,
-+	.max_rate = 2380800000,
- 	.min_rate = 940800000,
- 	.c = {
- 		.parent = &xo_a_clk.c,
-@@ -186,7 +186,7 @@ static struct pll_clk a53ss_sr_pll = {
- 		.config_ctl_val = 0x00141400,
- 	},
- 	.min_rate = 652800000,
--	.max_rate = 1478400000,
-+	.max_rate = 1708800000,
- 	.base = &virt_bases[APCS_C0_PLL_BASE],
- 	.c =  {
- 		.parent = &xo_a_clk.c,
-diff --git a/drivers/clk/qcom/clock-gcc-8976.c b/drivers/clk/qcom/clock-gcc-8976.c
-index 43828d7..2e01f43 100644
---- a/drivers/clk/qcom/clock-gcc-8976.c
-+++ b/drivers/clk/qcom/clock-gcc-8976.c
-@@ -1300,11 +1300,11 @@ static struct clk_freq_tbl ftbl_gfx3d_clk_src[] = {
- 	F( 266666667,          gpll0,    3,    0,     0),
- 	F( 300000000,    gpll4_gfx3d,    4,    0,     0),
- 	F( 366670000,          gpll3,    3,    0,     0),
--	F( 400000000,          gpll0,    2,    0,     0),
--	F( 432000000,    gpll6_gfx3d,  2.5,    0,     0),
--	F( 480000000,    gpll4_gfx3d,  2.5,    0,     0),
--	F( 550000000,          gpll3,    2,    0,     0),
--	F( 600000000,    gpll4_gfx3d,    2,    0,     0),
-+	F( 432000000,          gpll0,    2,    0,     0),
-+	F( 480000000,    gpll6_gfx3d,  2.5,    0,     0),
-+	F( 550000000,    gpll4_gfx3d,  2.5,    0,     0),
-+	F( 652800000,          gpll3,    2,    0,     0),
-+	F( 883200000,    gpll4_gfx3d,    2,    0,     0),
- 	F_END
- };
- 
-@@ -2564,8 +2564,8 @@ static struct branch_clk gcc_oxili_gfx3d_clk = {
- 		.dbg_name = "gcc_oxili_gfx3d_clk",
- 		.parent = &gfx3d_clk_src.c,
- 		VDD_DIG_FMAX_MAP5(LOWER, 300000000, LOW, 366670000,
--				NOMINAL, 432000000, NOM_PLUS, 480000000,
--				HIGH, 600000000),
-+				NOMINAL, 480000000, NOM_PLUS, 550000000,
-+				HIGH, 883200000),
- 		.ops = &clk_ops_branch,
- 		CLK_INIT(gcc_oxili_gfx3d_clk.c),
- 	},
diff --git a/arch/arm/boot/dts/qcom/kenzo/msm8956-kenzo.dtsi b/arch/arm/boot/dts/qcom/kenzo/msm8956-kenzo.dtsi
index 5677a0b..7db7d46 100644
--- a/arch/arm/boot/dts/qcom/kenzo/msm8956-kenzo.dtsi
+++ b/arch/arm/boot/dts/qcom/kenzo/msm8956-kenzo.dtsi
@@ -236,7 +236,7 @@
 			< 432000000   6 >,  /* NOM       */
 			< 480000000   7 >,  /* Turbo     */
 			< 550000000   8 >,  /* Super Turbo */
-			< 600000000   9 >;  /* Super Turbo */
+			< 883200000   9 >;  /* Super Turbo */
 		#clock-cells = <1>;
 	};
 
@@ -317,7 +317,7 @@
 			< 1305600000 6>,
 			< 1382400000 7>,
 			< 1401600000 8>,
-			< 1478400000 9>;
+			< 1708800000 9>;
 		qcom,speed0-bin-v0-c1 =
 			<          0 0>,
 			<  400000000 1>,
@@ -326,7 +326,7 @@
 			< 1382400000 4>,
 			< 1612800000 5>,
 			< 1747200000 6>,
-			< 1843200000 7>;
+			< 2380800000 7>;
 		qcom,speed0-bin-v0-cci =
 			<          0 0>,
 			<  307200000 1>,
@@ -344,7 +344,7 @@
 			< 1305600000 6>,
 			< 1382400000 7>,
 			< 1401600000 8>,
-			< 1478400000 9>;
+			< 1708800000 9>;
 		qcom,speed1-bin-v0-c1 =
 			<          0 0>,
 			<  400000000 1>,
@@ -353,7 +353,7 @@
 			< 1382400000 4>,
 			< 1612800000 5>,
 			< 1747200000 6>,
-			< 1843200000 7>;
+			< 2380800000 7>;
 		qcom,speed1-bin-v0-cci =
 			<          0 0>,
 			<  307200000 1>,
@@ -439,13 +439,13 @@
 				<  806400  4101 >,	/* SVS+  */
 				< 1017600  5712 >,	/* NOM   */
 				< 1190400  6152 >,	/* TURBO */
-				< 1478400  7104 >;
+				< 1708800  7104 >;
 			cpu-to-dev-map-4 =
 				<  883200   805 >,	/* SVS   */
 				< 1190400  4101 >,	/* SVS+  */
 				< 1382400  5712 >,	/* NOM   */
 				< 1612800  6152 >,	/* TURBO */
-				< 1843200  7104 >;
+				< 2380800  7104 >;
 		};
 
 		mincpubw-cpufreq {
@@ -453,7 +453,7 @@
 			cpu-to-dev-map-0 =
 				<  691200 2929 >,
 				< 1017600 2929 >,
-				< 1478400 4101 >;
+				< 1708800 4101 >;
 			cpu-to-dev-map-4 =
 				<  883200 2929 >,
 				< 1382400 2929 >,
@@ -467,13 +467,13 @@
 				<  806400  403200 >,
 				< 1017600  441600 >,
 				< 1190400  556800 >,
-				< 1478400  614400 >;
+				< 1708800  614400 >;
 			cpu-to-dev-map-4 =
 				<  883200 307200 >,
 				< 1190400 403200 >,
 				< 1382400 441600 >,
 				< 1612800 556800 >,
-				< 1843200 614400 >;
+				< 2380800 614400 >;
 		};
 	};
 
@@ -498,7 +498,8 @@
 			< 1305600 >,
 			< 1382400 >,
 			< 1401600 >,
-			< 1478400 >;
+			< 1478400 >,
+			< 1708800 >;
 		qcom,cpufreq-table-4 =
 			<  400000 >,
 			<  883200 >,
@@ -512,7 +513,8 @@
 			< 1382400 >,
 			< 1612800 >,
 			< 1747200 >,
-			< 1843200 >;
+			< 1843200 >,
+			< 2380800 >;
 	};
 
 	timer {
diff --git a/arch/arm/boot/dts/qcom/msm8956.dtsi b/arch/arm/boot/dts/qcom/msm8956.dtsi
index 36f9d774..1b53b20 100644
--- a/arch/arm/boot/dts/qcom/msm8956.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8956.dtsi
@@ -230,7 +230,7 @@
 			< 432000000   6 >,  /* NOM       */
 			< 480000000   7 >,  /* Turbo     */
 			< 550000000   8 >,  /* Super Turbo */
-			< 600000000   9 >;  /* Super Turbo */
+			< 883200000   9 >;  /* Super Turbo */
 		#clock-cells = <1>;
 	};
 
@@ -311,7 +311,7 @@
 			< 1305600000 6>,
 			< 1382400000 7>,
 			< 1401600000 8>,
-			< 1478400000 9>;
+			< 1708800000 9>;
 		qcom,speed0-bin-v0-c1 =
 			<          0 0>,
 			<  400000000 1>,
@@ -320,7 +320,7 @@
 			< 1382400000 4>,
 			< 1612800000 5>,
 			< 1747200000 6>,
-			< 1843200000 7>;
+			< 2380800000 7>;
 		qcom,speed0-bin-v0-cci =
 			<          0 0>,
 			<  307200000 1>,
@@ -338,7 +338,7 @@
 			< 1305600000 6>,
 			< 1382400000 7>,
 			< 1401600000 8>,
-			< 1478400000 9>;
+			< 1708800000 9>;
 		qcom,speed1-bin-v0-c1 =
 			<          0 0>,
 			<  400000000 1>,
@@ -347,7 +347,7 @@
 			< 1382400000 4>,
 			< 1612800000 5>,
 			< 1747200000 6>,
-			< 1843200000 7>;
+			< 2380800000 7>;
 		qcom,speed1-bin-v0-cci =
 			<          0 0>,
 			<  307200000 1>,
@@ -381,6 +381,7 @@
 	};
 
 	cpubw: qcom,cpubw {
+		label = "cpubw";
 		compatible = "qcom,devbw";
 		governor = "cpufreq";
 		qcom,src-dst-ports = <1 512>;
@@ -398,6 +399,7 @@
 	};
 
 	mincpubw: qcom,mincpubw {
+		label = "mincpubw";
 		compatible = "qcom,devbw";
 		governor = "cpufreq";
 		qcom,src-dst-ports = <1 512>;
@@ -431,13 +433,13 @@
 				<  806400  4101 >,	/* SVS+  */
 				< 1017600  5712 >,	/* NOM   */
 				< 1190400  6152 >,	/* TURBO */
-				< 1478400  7104 >;
+				< 1708800  7104 >;
 			cpu-to-dev-map-4 =
 				<  883200   805 >,	/* SVS   */
 				< 1190400  4101 >,	/* SVS+  */
 				< 1382400  5712 >,	/* NOM   */
 				< 1612800  6152 >,	/* TURBO */
-				< 1843200  7104 >;
+				< 2380800  7104 >;
 		};
 
 		mincpubw-cpufreq {
@@ -445,7 +447,7 @@
 			cpu-to-dev-map-0 =
 				<  691200 2929 >,
 				< 1017600 2929 >,
-				< 1478400 4101 >;
+				< 1708800 4101 >;
 			cpu-to-dev-map-4 =
 				<  883200 2929 >,
 				< 1382400 2929 >,
@@ -459,13 +461,13 @@
 				<  806400  403200 >,
 				< 1017600  441600 >,
 				< 1190400  556800 >,
-				< 1478400  614400 >;
+				< 1708800  614400 >;
 			cpu-to-dev-map-4 =
 				<  883200 307200 >,
 				< 1190400 403200 >,
 				< 1382400 441600 >,
 				< 1612800 556800 >,
-				< 1843200 614400 >;
+				< 2380800 614400 >;
 		};
 	};
 
@@ -490,7 +492,8 @@
 			< 1305600 >,
 			< 1382400 >,
 			< 1401600 >,
-			< 1478400 >;
+			< 1478400 >,
+			< 1708800 >;
 		qcom,cpufreq-table-4 =
 			<  400000 >,
 			<  883200 >,
@@ -504,7 +507,8 @@
 			< 1382400 >,
 			< 1612800 >,
 			< 1747200 >,
-			< 1843200 >;
+			< 1843200 >,
+			< 2380800 >;
 	};
 
 	timer {
diff --git a/arch/arm/boot/dts/qcom/msm8976-gpu.dtsi b/arch/arm/boot/dts/qcom/msm8976-gpu.dtsi
index 31429fa..6fc9df9 100644
--- a/arch/arm/boot/dts/qcom/msm8976-gpu.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8976-gpu.dtsi
@@ -132,7 +132,7 @@
 			/* TURBO */
 			qcom,gpu-pwrlevel@0 {
 				reg = <0>;
-				qcom,gpu-freq = <600000000>;
+				qcom,gpu-freq = <883200000>;
 				qcom,bus-freq = <10>;
 				qcom,bus-min = <9>;
 				qcom,bus-max = <10>;
diff --git a/arch/arm/boot/dts/qcom/msm8976-regulator.dtsi b/arch/arm/boot/dts/qcom/msm8976-regulator.dtsi
index 96c9950..ac2b36e 100644
--- a/arch/arm/boot/dts/qcom/msm8976-regulator.dtsi
+++ b/arch/arm/boot/dts/qcom/msm8976-regulator.dtsi
@@ -453,7 +453,7 @@
 				<6  1305600000>,
 				<7  1382400000>,
 				<8  1401600000>,
-				<9  1478400000>;
+				<9  1708800000>;
 		qcom,cpr-speed-bin-max-corners =
 				<0 0 2 4 8>,
 				<1 0 2 4 9>;
@@ -548,7 +548,7 @@
 				<4  1382400000>,
 				<5  1612800000>,
 				<6  1747200000>,
-				<7  1843200000>;
+				<7  2380800000>;
 		qcom,cpr-speed-bin-max-corners =
 				<0 0 2 4 6>,
 				<1 0 2 4 7>;
diff --git a/drivers/clk/qcom/clock-cpu-8976.c b/drivers/clk/qcom/clock-cpu-8976.c
index 130314d..4690a2e 100644
--- a/drivers/clk/qcom/clock-cpu-8976.c
+++ b/drivers/clk/qcom/clock-cpu-8976.c
@@ -141,7 +141,7 @@ static struct pll_clk a72ss_hf_pll = {
 		.vco_mode_masked = BVAL(21, 20, 1),
 	},
 	.base = &virt_bases[APCS_C1_PLL_BASE],
-	.max_rate = 1843200000,
+	.max_rate = 2380800000,
 	.min_rate = 940800000,
 	.c = {
 		.parent = &xo_a_clk.c,
@@ -186,7 +186,7 @@ static struct pll_clk a53ss_sr_pll = {
 		.config_ctl_val = 0x00141400,
 	},
 	.min_rate = 652800000,
-	.max_rate = 1478400000,
+	.max_rate = 1708800000,
 	.base = &virt_bases[APCS_C0_PLL_BASE],
 	.c =  {
 		.parent = &xo_a_clk.c,
diff --git a/drivers/clk/qcom/clock-gcc-8976.c b/drivers/clk/qcom/clock-gcc-8976.c
index 43828d7..818db9f 100644
--- a/drivers/clk/qcom/clock-gcc-8976.c
+++ b/drivers/clk/qcom/clock-gcc-8976.c
@@ -1304,7 +1304,7 @@ static struct clk_freq_tbl ftbl_gfx3d_clk_src[] = {
 	F( 432000000,    gpll6_gfx3d,  2.5,    0,     0),
 	F( 480000000,    gpll4_gfx3d,  2.5,    0,     0),
 	F( 550000000,          gpll3,    2,    0,     0),
-	F( 600000000,    gpll4_gfx3d,    2,    0,     0),
+	F( 883200000,    gpll4_gfx3d,    2,    0,     0),
 	F_END
 };
 
@@ -2565,7 +2565,7 @@ static struct branch_clk gcc_oxili_gfx3d_clk = {
 		.parent = &gfx3d_clk_src.c,
 		VDD_DIG_FMAX_MAP5(LOWER, 300000000, LOW, 366670000,
 				NOMINAL, 432000000, NOM_PLUS, 480000000,
-				HIGH, 600000000),
+				HIGH, 883200000),
 		.ops = &clk_ops_branch,
 		CLK_INIT(gcc_oxili_gfx3d_clk.c),
 	},
